bindings/java/src/org/sleuthkit/datamodel/Report.java

```diff
@@ -18,49 +18,74 @@
  */
 package org.sleuthkit.datamodel;
 
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import static java.nio.file.StandardOpenOption.READ;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 /**
- * This is a data transfer object (DTO) class that models reports.
+ * This is a class that models reports.
  */
-public class Report {
+public class Report implements Content {
 
 	static long ID_NOT_SET = -1;
 	private long id = ID_NOT_SET;
-	private final String path;
+	private final Path path;
```

Comment:
Given that the id field is now a Sleuthkit object id, it would be nice to rename it objId or objectId and also deprecate getId(), replacing it with getObjectId(). I recall that when I wrote this class originally, I was trying to make it clear that this property was not an object id, which, of course, it now is.  We could make the column in the reports table be obj_id during the upgrade, perhaps.

Reply:
I can make the change to the Report class but in a previous conversation we decided not to change the name of the column in the reports table for backwards compatibility purposes (old Autopsy versions will be able to open upgraded cases).

Reply:
This is an implementation of Content.getId() so creating new method and deprecating old doesn't seem like a good idea.

Reply:
Sorry, I forgot about that decision. Please let it be, then.